src/assets/scripts/client/scope/RadarTargetCollection.js

```diff
@@ -0,0 +1,214 @@
+ *
+ * @class RadarTargetCollection
+ */
+export default class RadarTargetCollection extends BaseCollection {
+    constructor(theme) {
+        super();
+
+        this._eventBus = EventBus;
+        this._theme = theme;
+
+        this._init()
+            .enable();
+    }
+
+    /**
+     * Return a read-only reference to the collection array
+     *
+     * @for RadarTargetCollection
+     * @property items
+     * @type {array}
+     */
+    get items() {
+        return this._items;
+    }
+
+    /**
+     * Complete initialization tasks
+     *
+     * @for RadarTargetCollection
+     * @method _init
+     * @param theme {object}
+     * @private
+     * @chainable
+     */
+    _init() {
+        return this;
+    }
+
+    /**
+     * Add the provided `RadarTargetModel` instance to the collection
+     *
+     * @for RadarTargetCollection
+     * @method addRadarTargetModel
+     * @param radarTargetModel {RadarTargetModel}
+     */
+    addRadarTargetModel(radarTargetModel) {
+        if (!(radarTargetModel instanceof RadarTargetModel)) {
+            throw new TypeError(`Expected instance of RadarTargetModel but received '${radarTargetModel}'`);
+        }
+
+        this._items.push(radarTargetModel);
+    }
+
+    /**
+     * Create `RadarTargetModel`s for the given `AircraftModel`
+     *
+     * @for RadarTargetCollection
+     * @method addRadarTargetModelFromAircraftModel
+     * @param aircraftModel {AircraftModel}
+     */
+    addRadarTargetModelFromAircraftModel = (aircraftModel) => {
+        const radarTargetModel = new RadarTargetModel(this._theme, aircraftModel);
+
+        this.addRadarTargetModel(radarTargetModel);
+    };
+
+    /**
+     * Activate event handlers
+     *
+     * @for RadarTargetModel
+     * @method disable
+     */
+    disable() {
+        this._eventBus.off(EVENT.ADD_AIRCRAFT, this.addRadarTargetModelFromAircraftModel);
+        this._eventBus.off(EVENT.REMOVE_AIRCRAFT, this.removeRadarTargetModel);
+        this._eventBus.off(EVENT.SET_THEME, this._setTheme);
+    }
+
+    /**
+     * Deactivate event handlers
+     *
+     * @for RadarTargetModel
+     * @method disable
+     */
+    enable() {
+        this._eventBus.on(EVENT.ADD_AIRCRAFT, this.addRadarTargetModelFromAircraftModel);
+        this._eventBus.on(EVENT.REMOVE_AIRCRAFT, this.removeRadarTargetModel);
+        this._eventBus.on(EVENT.SET_THEME, this._setTheme);
+    }
+
+    /**
+     * Get the radar target model object for the specified aircraft
+     *
+     * @for RadarTargetCollection
+     * @method getRadarTargetModelFromAircraftModel
+     * @param aircraftModel {AircraftModel}
+     * @return radarTargetModel {RadarTargetModel}
+     */
+    getRadarTargetModelFromAircraftModel(aircraftModel) {
+        // Store variable because `this` within lodash `_filter` has different scope
+        const radarTargetModels = this._items;
+        const results = _filter(radarTargetModels, (radarTargetModel) =>
```

Comment:
really? that seems gross 😕

whateva! done :)

Reply:
it is gross, I agree.  but because you're using a direct return, its preferable to have that all on the same line.  Otherwise some other dev in the future could get confused and forget that it's a direct return (like me 😊)